src/useMeasure/index.ts

```diff
@@ -12,9 +12,11 @@ export type Measures = {
  * Uses ResizeObserver to track element dimensions and re-render component when they change.
  *
  * @param enabled Whether resize observer is enabled or not.
+ * @param observerEntryMatcher A custom matcher function to define the measures. If returns `null` - default `contentRect` measures are applied
  */
 export function useMeasure<T extends Element>(
 	enabled = true,
+	observerEntryMatcher?: (entry: ResizeObserverEntry) => Measures | null,
```

Comment:
Proper solution here would be:
1) to define the type `type Measurer = (entry: ResizeObserverEntry) => Measures`. note there is no option for measure to return `null` as you did, since in case element exists it is measurable (a don't see option for etry to me non-measurable)
2) expose two implementations of measurer - `contentBoxMeasurer` and `borderBoxMeasurer`
3) add parameter that defaults to `contentBoxMeasurer`
4) use that measurer all the time